modules/cudaarithm/include/opencv2/cudaarithm.hpp

```diff
@@ -358,6 +358,31 @@ threshold types are not supported.
  */
 CV_EXPORTS_W double threshold(InputArray src, OutputArray dst, double thresh, double maxval, int type, Stream& stream = Stream::Null());
 
+/** @brief  Checks if array elements lie between two scalars.
+
+The function checks the range as follows:
+-   For every element of a single-channel input array:
+    \f[\texttt{dst} (I)= \texttt{lowerb}_0  \leq \texttt{src} (I)_0 \leq  \texttt{upperb}_0\f]
+-   For two-channel arrays:
+    \f[\texttt{dst} (I)= \texttt{lowerb}_0  \leq \texttt{src} (I)_0 \leq  \texttt{upperb}_0  \land \texttt{lowerb}_1  \leq \texttt{src} (I)_1 \leq  \texttt{upperb}_1\f]
+-   and so forth.
+
+That is, dst (I) is set to 255 (all 1 -bits) if src (I) is within the
+specified 1D, 2D, 3D, ... box and 0 otherwise.
+
+Note that unlike the CPU inRange, this does NOT accept an array for lowerb or
+upperb, only a cv::Scalar.
+
+@param src first input array.
+@param lowerb inclusive lower boundary cv::Scalar.
+@param upperb inclusive upper boundary cv::Scalar.
+@param dst output array of the same size as src and CV_8U type.
+@param stream Stream for the asynchronous version.
+
+@sa inRange
+ */
+CV_EXPORTS_W void inRange(InputArray src, InputArray lowerb, InputArray upperb, OutputArray dst, Stream& stream = Stream::Null());
```

Comment:
> CV_EXPORTS_W

Did you try to call this from Python binding? Are `lowerb` / `upperb` handled properly?

Reply:
I can make calls like
```python
src = (np.random.random((1920, 1080, 4)) * 256).astype(np.uint8)
lowerb = (0, 0, 0, 0)
upperb = (255, 255, 255, 255)
dst = cv2.cuda.inRange(src, lowerb, upperb)
```
with different types and numbers of channels for `src` (up to 4 channels).  The `CV_Check` fires correctly if I pass the wrong shape for `lowerb` or `upperb`, or too many channels for `src`.  It doesn't work if I pass a `cv2.cuda_GpuMat` (I get a `TypeError: Expected Ptr<cv::UMat> for argument 'src'`) - I'm really not familiar with using OpenCV CUDA from Python so I don't know if that's the intended behavior.

Reply:
> I get a TypeError

Current messages from Python bindings are quite useless. You can set `OPENCV_PYTHON_DEBUG=1` env variable to investigate conversion errors.
The "GPU" overload assumes that all inputs must be a `GpuMat` including `lowerb/upperb` parameters (because they a `InputArray`). But wrapping 4 scalars into `GpuMat` has significant overhead.

> InputArray lowerb, InputArray upperb

Try to replace type to `const Scalar&` to eliminate unnecessary complexity.

Reply:
Please add simple Python test into this [file](https://github.com/opencv/opencv_contrib/blob/master/modules/cudaarithm/misc/python/test/test_cudaarithm.py).

(please note, that public OpenCV CI doesn't run CUDA code/tests)

Reply:
Done.  I switched to `const Scalar&` for `lowerb` and `upperb`, and added a Python test to that file which passes on my machine, it now works if you pass a `Mat` or a `cuda_GpuMat`